x/mongo/driver/operation.go

```diff
@@ -457,7 +458,9 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 			operationErr.Labels = tt.Labels
 		case Error:
 			if tt.HasErrorLabel(TransientTransactionError) || tt.HasErrorLabel(UnknownTransactionCommitResult) {
-				op.Client.ClearPinnedServer()
+				if op.Client != nil {
```

Comment:
nit: this one isn't actually necessary, ClearPinnedServer checks if the client it's called on is nil

Reply:
Ah didn't look in the `ClearPinnedServer()` and `UpdateRecoveryToken()` functions. Thank you :) Done.